src/cargo/core/resolver/dep_cache.rs

```diff
@@ -513,25 +515,60 @@ impl RequirementError {
                     .filter(|dep| dep.name_in_toml() == feat)
                     .collect();
                 if deps.is_empty() {
+                    let closest =
+                        closest_msg(&feat.as_str(), summary.features().keys(), |key| &key);
```

Comment:
How expensive is the `closest_msg` call? and same for the coding added below?

This part of the code can be hit very often in a build. So operations like an allocation should be done only where absolutely necessary, if possible only in the branch where `ActivateError::Fatal` is returned. Happy to eat the performance hit if it's necessary for better error messages. But, if we can avoid it I would like to.

Reply:
Is there any happy-path performance hit if this is limited to `Fatal` branches?

Reply:
The changes are now only limited to the `Fatal` branches, I believe

Reply:
Thank you so much for the change.